app/scenes/messagesScene.js

```diff
@@ -50,14 +51,20 @@ export default class MessagesScene extends BaseScene<{}> {
         <Modal
           animationType='fade'
           visible={this.state.showMessageDetailModal}
-          onRequestClose={() => this.closeMessageDetailModal()}>
-          <View>
-            <Text>Modal for messages</Text>
-          </View>
+          onRequestClose={() => this.closeMessageDetailModal()}
+        >
           <Button
+            color='#70B5E5'
             onPress={() => this.closeMessageDetailModal()}
             title="Close Modal"
           />
+          <View style={stylesModal}>
+            <Text style={stylesDateCreated}>Primljeno: {Moment(this.state.modalItem.Created).format('DD.MM.YYYY.')}</Text>
+            <View>
+              <Text style={stylesSubjectText}>{this.state.modalItem.Subject}</Text>
+            </View>
+            <Text style={stylesModalText}>{this.state.modalItem.Text}</Text>
+          </View>
         </Modal>
 
         <FlatList
```

Comment:
Za FlatList komponentu proslijedi this.state kao extraData atribut kako bi se scena rerenderala.

```
<FlatList
          contentContainerStyle={stylesFlatList}
          numColumns={1}
          data={this.state.messages}
          extraData={this.state}
```

https://facebook.github.io/react-native/docs/flatlist.html

> This is a PureComponent which means that it will not re-render if props remain shallow- equal. Make sure that everything your renderItem function depends on is passed as a prop (e.g. extraData) that is not === after updates, otherwise your UI may not update on changes. This includes the data prop and parent component state.